cmd/lifecycle/restorer.go

```diff
@@ -32,12 +47,27 @@ func (r *restoreCmd) DefineFlags() {
 	cmd.FlagLayersDir(&r.layersDir)
 	cmd.FlagUID(&r.uid)
 	cmd.FlagGID(&r.gid)
+	if api.MustParse(r.platformAPI).Compare(api.MustParse("0.6")) >= 0 { // platform API >= 0.6
+		cmd.FlagUseDaemon(&r.useDaemon)
+		cmd.FlagSkipLayers(&r.skipLayers)
+	}
 }
 
 func (r *restoreCmd) Args(nargs int, args []string) error {
-	if nargs > 0 {
-		return cmd.FailErrCode(errors.New("received unexpected Args"), cmd.CodeInvalidArgs, "parse arguments")
+	if api.MustParse(r.platformAPI).Compare(api.MustParse("0.6")) < 0 { // platform API < 0.6
```

Comment:
Not to throw a wrench in this feature, but given the complexity involved I wonder if it's worth trying to add a few high-level acceptance tests around restorer.

Reply:
Yeah - I was sort of afraid someone might say that 😬 

I don't mind throwing some time at some acceptance tests. I may attempt to do it on a fresh branch off of main and then bring it in to this branch once we are happy. Just in case it fails, I don't want it clogging up this work.

Reply:
So happy to see https://github.com/buildpacks/lifecycle/pull/512 :)